astropy/timeseries/tests/test_downsample.py

```diff
@@ -194,3 +194,35 @@ def test_downsample_edge_cases(time, time_bin_start, time_bin_end):
         assert down['a'].mask.all()        # all bins placed *beyond* the time span of the data
     elif ts.time.min() < time_bin_start[1]:
         assert down['a'][0] == ts['a'][0]  # single-valued time series falls in *first* bin
+
+
+@pytest.mark.parametrize("diff_from_base",
+                         [1 * u.year, 10 * u.year, 50 * u.year, 100 * u.year])
+def test_time_precision_limit(diff_from_base):
```

Comment:
```suggestion
@pytest.mark.parametrize("scale",
                         [1, 10, 50, 100])
def test_time_precision_limit(scale):
```
Might set scaled limits (see below).